miner/multi_worker.go

```diff
@@ -86,14 +86,60 @@ func (w *multiWorker) disablePreseal() {
 	}
 }
 
+// buildPayload builds the payload according to the provided parameters.
+func (w *multiWorker) payloadAssembler(args *BuildPayloadArgs) (*Payload, error) {
```

Comment:
curious if this is copied from somewhere?

Reply:
it is built on the template of `buildPayload` dude. the major changes are in `sealingBlock`